inst/private/python_ipc_popen2.m

```diff
@@ -43,11 +44,29 @@
   info = [];
 
   if (strcmp(what, 'reset'))
-    cleanup = [];  % triggers a call to reset function
+    if (~ isempty (pid))
+      if (verbose)
+        disp ('Closing the Python communications link.')
+      end
+    end
+    A = true;
+    if (~ isempty (fin))
+      % produces a single newline char: not sure why
+      t = fclose (fin);
+      fin = [];
+      waitpid (pid);
+      pid = [];
+      A = A && (t == 0);
+    end
+    if (~ isempty (fout))
```

Comment:
Should we change `~ isempty (fin)` and `~ isempty (fout)` to `is_valid_file_id (fin)` and `is_valid_file_id (fout)` respectively?

Reply:
good point!  That does seem safer if the onCleanup somehow triggered *before* this code!